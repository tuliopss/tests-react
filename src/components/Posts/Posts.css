.posts {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 30px;
}
.post img {
  max-width: 100%;
}
.post-content {
  padding: 30px;
}
