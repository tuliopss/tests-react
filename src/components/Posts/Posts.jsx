import PostCard from "../PostCard/PostCard";
import "./Posts.css";
const Posts = ({ posts = [] }) => (
  <div className='posts'>
    {posts.map((post) => (
      <PostCard
        key={post.id}
        title={post.title}
        body={post.body}
        id={post.id}
        cover={post.cover}
      />
    ))}
  </div>
);
export default Posts;
