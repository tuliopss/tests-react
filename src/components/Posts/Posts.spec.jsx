import { describe, expect, it } from "vitest";
import Posts from "./Posts";
import { render, screen } from "@testing-library/react";

const propsMock = {
  posts: [
    {
      id: 1,
      title: "title",
      body: "body 1",
      cover: "img/img.png",
    },
    {
      id: 2,
      title: "title 2",
      body: "body 2",
      cover: "img/img.png 2",
    },
    {
      id: 3,
      title: "title 3",
      body: "body 3",
      cover: "img/img.png 3",
    },
  ],
};
describe("<Posts/>", () => {
  it("should render posts", () => {
    render(<Posts {...propsMock} />);

    expect(screen.getAllByRole("heading", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole("img", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole("img", { name: /title 3/i })).toHaveAttribute(
      "src",
      "img/img.png 3"
    );
  });

  it("should not render posts", () => {
    render(<Posts />);

    expect(
      screen.queryByRole("heading", { name: /title/i })
    ).not.toBeInTheDocument();  
  });

  it("should match snapshot", () => {
    const { container } = render(<Posts {...propsMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
