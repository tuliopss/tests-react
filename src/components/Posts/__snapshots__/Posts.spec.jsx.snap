// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Posts/> > should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title
         
        1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="img/img.png 2"
    />
    <div
      class="post-content"
    >
      <h2>
        title 2
         
        2
      </h2>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="img/img.png 3"
    />
    <div
      class="post-content"
    >
      <h2>
        title 3
         
        3
      </h2>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
